src/range/bsearch.cr

```diff
@@ -99,6 +105,25 @@ struct Range(B, E)
       end
     {% end %}
 
+    {% if B < Number %}
```

Comment:
IMO this seems too general because it'll effectively catch `Int::Primitive | BigInt | BigFloat | BigRational | BigDecimal`. (There is no `Float::Primitive` because of the above if-expressions.) I think only the integers are applicable here since the rest of them have arbitrarily high precision